es-ds-components/components/es-data-table.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+    <DataTable
+        scrollable
+        :pt="{
+            table: {
+                class: 'font-size-75 font-size-md-100',
+                style: 'border-collapse: separate; border-spacing: 0;'
+            },
+            wrapper: { 
+                style: 'border-radius: inherit'
+            },
+            thead:{
+                style: 'top: 0px; z-index: 2'
+            }
+        }"
+        :value="props.items"
+        class="border table__data-table table-spacing table"
+        :scrollHeight="scrollHeight"
+    >
+    <Column v-for="(col,index) of columns" :key="col?.key ? col.key : col" :field="col?.key ? col.key : col" :header="label(col)" 
+        :sortable="col?.sortable"
+        :frozen="checkFrozenRow(index)" alignFrozen="left"
+        :pt="{
+            sortIcon:{
+                style: 'display: none'
+            }, 
+            headerCell: (options)  => ({
+                class:[
+                    {
+                        'stickyFirstCol': options.props.frozen,
+                    }
+                ],
+            }),
+            bodyCell: (options)  => ({
+                class:[
+                    {
+                        'stickyFirstCol': options.props.frozen,
+                    }
+                ],
+            }),  
+        }"
+        >
+    </Column>
+    </DataTable>
+</template>
+<script setup lang="ts">
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+const props = defineProps({
+    fields: {
+        type: Array,
+        default: null,
+    },
+    items: {
+        type: Array,
+        required: true,
+    },
+    stickyFirstCol: {
+        type: Boolean,
+        default: false,
+    },
+    stickyHeader: {
+        type: [Boolean, String],
+        default: false,
+    },
+});
+
+const checkFrozenRow = (index) => {
+    if (index === 0 && props.stickyFirstCol) {
+        return true;
+    }
+    return false;
+};
+
+const label = (col) => {
+    if (typeof col == 'string') {
+        return convertKey(col);
+    }
+    return col?.label? col.label : convertKey(col?.key);
+};
+
+// Convert camel case and make first letter of string uppercase
+const convertKey = (string) => {
+    if (typeof string == 'string') {
+        const stringSpaced = string.replace(/([a-z])([A-Z])/g, '$1 $2');
+        return stringSpaced.charAt(0).toUpperCase() + stringSpaced.slice(1);
+    }
+    return string;  
```

Comment:
> Implemented uppercasing and converting camel case keys for the example table and ArticleDataTable [example](https://github.com/EnergySage/es-cms-articles/blob/4f94808e0fc1c33525597f7a075f685b24bdd5a8/components/storyblok/ArticleDataTable.vue#L42C16-L42C80), but b-table also converts kebab-case and snake_case to individual words

Is there any reason not to just copy BootstrapVue's logic for that into here?

Reply:
Changed to the code to use bootstraps logic!